Documentation/scalability/scalability.rst

```diff
@@ -0,0 +1,397 @@
+used 250 different CNP objects, to correctly measure the convergence time we
+could have used a single CNP with multiple policy rules defined under the
+``specs`` field (not ``spec``) for a CNP.
+
+However we can see the time it took the last Cilium agent to increment its
+Policy Revision, which is incremented individually on each Cilium agent every
+time time a CiliumNetworkPolicy (CNP) is received, between second ``15:45:44``
+and ``15:45:46`` and see when was the last time an Endpoint was regenerated by
+checking the 99th percentile of the "Endpoint regeneration time", we can see
+that it took less than 5s. We can also verify **the maximum time was less than
+``600ms`` for an endpoint to have the policy enforced.**
+
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+9. Deploy 250 policies for CiliumClusterwideNetworkPolicies (CCNP)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The difference between these policies and the previous ones installed is that
+these ones select all pods in all namespaces. To recap, this means that we will
+now have **250 different network policies selecting 10000 pods and 250 different
+network policies selecting 50000 pods on a cluster with 1000 nodes.** Similarly
+to the previous step we will deploy 125 L4 policies and another 125 L7 policies.
+
+.. figure:: images/image_9_01.png
+.. figure:: images/image_9_02.png
+
+Similar to the creation of the previous 250 CNPs, there was also an increase in
+CPU usage during the creation of the CCNPs. The CPU usage was similar even
+though the policies were effectively selecting more pods.
+
+.. figure:: images/image_9_03.png
+
+As all pods running in a node are selected by **all the 250 CCNPs created**, we
+see an increase of the **Endpoint regeneration time** which **peaked a little
+above 3s.**
+
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+10. "Accidentally" destroy 10000 pods
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In this step we have "accidentally" destroyed 10000 random pods. Kubernetes will
+then recreate 10000 new pods so it will help us understand what is the
+convergence time for all the network policies deployed.
+
+.. figure:: images/image_10_01.png
+.. figure:: images/image_10_02.png
+
+
+* The first mark represents the point in time when pods were "deleted" and the
+  second mark represents the point in time 10k new pods finished to be
+  recreated.
+
+* Besides the CPU usage slightly increasing while pods are being scheduled in
+  the cluster we did see some interesting data points in the BPF memory usage.
+  As each endpoint can have one or more dedicated BPF maps, the BPF memory usage
+  is directly correlated with the number of pods running in a node. **If the
+  number of pods per node decreases so does the BPF memory usage.**
+
+.. figure:: images/image_10_03.png
+
+Another interesting data point is the Endpoint regeneration time is "inexistent".
```

Comment:
not clear what this sentence means. does this mean that the prometheus metric for endpoint regeneration was missing? maybe omit the first 2 sentences and say something like:

We inferred the time it took for all the endpoints to get regenerated by looking at the number of Cilium endpoints with the policy enforced over time.